### DISCUSSION TITLE
Implications of disabling XA transactions (transaction-xa-enabled=false)

### DISCUSSION THREAD

**Discussion by myoffe** [keycloak/keycloak#26873] [Q&A]
Hi,

I want to use Keycloak with the [AWS JDBC Driver](https://github.com/awslabs/aws-advanced-jdbc-wrapper), which doesn't appear to support XA transactions. The [docs](https://www.keycloak.org/server/db#_using_database_vendors_without_xa_transaction_support) state:

> Keycloak uses XA transactions and the appropriate database drivers by default. Certain vendors, such as Azure SQL and MariaDB Galera, do not support or rely on the XA transaction mechanism. To use Keycloak without XA transaction support using the appropriate JDBC driver, enter the following command:

What are the downsides or possible implications of disabling XA transactions with keycloak?

Thanks! 🙏

**Comment by sebix** [comment]
As far as I found out, XA transactions are only useful in distributed setups with multiple databases. For this reason, XA transactions are now disabled by default.

https://github.com/keycloak/keycloak/issues/27308 contains a lengthy discussion by the devs with some details.
